### PR TITLE
Tls profiler testing

### PR DESCRIPTION
**Pull request by jacoba109**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [ ] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [ ] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

################################################################################################

Adds unit and FAT tests for tlsProfiler

################################################################################################

### CODE REVIEW

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 1]
```diff
@@ -0,0 +1,405 @@
+/*******************************************************************************
+ * Copyright (c) 2026 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:9]: I think these 15 tests are too much; we try to limit it to about 6 tests and keep maybe the main positive and negative ones as LITE and the rest as FULL.

**Comment by rangaran** [comment]
A couple of action items from our last discussion:
- Adding global variables
- Adding more meaningful tests, like for TLS1.0 usage
- Remove hard-coded areas and make functions dependent on an input rather than hardcoding that

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 1 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:9]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+        securityUtilityPath = libertyInstallRoot + "/bin/securityUtility";
+        testMachine = server.getMachine();
+        httpsPort = server.getHttpDefaultSecurePort();
+
+        // Point the securityUtility JVM at the Liberty server's auto-generated keystore so
+        // that TLS handshakes with the self-signed certificate succeed.  The keystore is
+        // guaranteed to exist at this point because assertServerReady() already waited for
+        // CWPKI0803A (the "SSL certificate created" message).
+        //
+        // getServerRoot() returns the canonical server directory, which is portable across
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: no need for so many comments. 
clean up the file a bit

File: dev/com.ibm.ws.security.utility_fat/publish/files/TLSProfilerTestServer/serverTLSv11.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        <feature>requestTiming-1.0</feature>
+        <feature>transportSecurity-1.0</feature>
+    </featureManager>
+
+    <keyStore id="defaultKeyStore" password="Liberty"/>
+
+    <ssl id="defaultSSLConfig" trustDefaultCerts="true" sslProtocol="TLSv1.1"/>
+
+    <logging traceSpecification="*=info:com.ibm.ws.security.token.ltpa.pqc.*=all:com.ibm.ws.security.audit.*=all:*=info:SSLChannel=all:com.ibm.ws.ssl.*=all:com.ibm.websphere.ssl=all:com.ibm.wsspi.ssl.*=all"
+       traceFileName="ssl_trace.log"
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/publish/files/TLSProfilerTestServer/serverTLSv11.xml]: Remove the ssl tracing in the logs (it's just for debugging) - generally not enabled for final fat code.

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+                    new String[] { "tlsProfiler", "--host=" + HOST, "--port=" + certServerPort },
+                    libertyInstallRoot, new Properties(), SHORT_TIMEOUT_SECS);
+
+            Log.info(thisClass, testName.getMethodName(), "running on localhost: " + certServerPort);
+
+            Log.info(thisClass, testName.getMethodName(),
+                     "RC: " + output.getReturnCode() + "\nstdout:\n" + output.getStdout() + "\nstderr:\n" + output.getStderr());
+
+            assertNotNull("Server should log CWWKO0801E with 'certificate_unknown' when the profiler connects without a trusted certificate",
+                          untrustedCertServer.waitForStringInLogUsingMark("CWWKO0801E.*certificate_unknown", 30000));
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: try searching for something not in the ssl_trace which would indicate the cert unknown.

File: dev/com.ibm.ws.security.utility_fat/publish/servers/TLSProfilerUntrustedCertServer/server.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<server description="TLS Profiler FAT — untrusted certificate test server">
+    <include location="../fatTestCommon.xml"/>
+    <featureManager>
+        <feature>jsp-2.3</feature>
+        <feature>appSecurity-3.0</feature>
+        <feature>requestTiming-1.0</feature>
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/publish/servers/TLSProfilerUntrustedCertServer/server.xml]: is request timing necessary?

Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/publish/servers/TLSProfilerUntrustedCertServer/server.xml]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 4 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Done

File: dev/com.ibm.ws.security.utility_fat/publish/files/TLSProfilerTestServer/serverTLSv11.xml (hunk 3 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/publish/files/TLSProfilerTestServer/serverTLSv11.xml]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 2 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Done

**Comment by jacoba109** [comment]
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=4f3af742-66ca-4879-8aff-2a088df03f80) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

**Comment by jacoba109** [comment]
Added tests for full functionality of TLSProfiler with positive and negative cases, including edge cases regarding different TLS versions, argument parsing, and unreachable ports.

File: dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java [hunk 6]
```diff
@@ -12,9 +12,16 @@
  *******************************************************************************/
 package com.ibm.ws.security.utility.tasks;
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
+import java.io.IOException;
 import java.io.PrintStream;
+import java.net.ServerSocket;
+import java.net.Socket;
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java]: Why are there changes in this file?

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+    //--------------------------------------------------------------------------
+    // Tests
+    //--------------------------------------------------------------------------
+
+    /**
+     * Validates that the profiler output accurately tracks the server's active TLS
+     * version across a live configuration swap, and that outdated/unsafe TLS versions (TLSv1.1) fail
+     */
+    @Test
+    @Mode(TestMode.LITE)
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: make this full

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+                   fileContent.contains("TLSv1.3:"));
+        numberedFile.delete();
+    }
+
+    /**
+     * Each required argument ({@code --host}, {@code --port}) is independently
+     * mandatory; omitting either one must cause a non-zero exit code.
+     */
+    @Test
+    @Mode(TestMode.FULL)
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: lite not full

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+        ProgramOutput output = runTlsProfiler("--verbose");
+        assertEquals(SUCCESS_RC, output.getReturnCode());
+        assertTrue("--verbose stdout should contain successful-handshakes section",
+                   output.getStdout().contains("Successful handshakes"));
+        assertTrue("--verbose stdout should contain unsuccessful-handshakes section",
+                   output.getStdout().contains("Unsuccessful handshakes"));
+    }
+
+    /**
+     * When {@code --file} is given, the profiler writes results to that path and
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: explain better

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+    }
+
+    /**
+     * {@code --verbose} must produce both the successful-handshakes and the
+     * unsuccessful-handshakes sections.
+     */
+    @Test
+    @Mode(TestMode.FULL)
+    public void testVerboseOutputReportsUnsuccessfulHandshakes() throws Exception {
+        ProgramOutput output = runTlsProfiler("--verbose");
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:259]: lite

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 11]
```diff
@@ -0,0 +1,388 @@
...
+        } finally {
+            untrustedCertServer.stopServer("CWWKO0801E");
+        }
+    }
+
+    /**
+     * Pointing at a port on which nothing is listening causes the profiler to
+     * fail on the connect attempt and exit non-zero.
+     */
+    @Test
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:336]: full not lite

**Review by rangaran — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 10 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:259]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 11 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:336]: Done

File: dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java (hunk 6 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java]: Had previously added tests, then we decided against them and I did not properly roll back the changes. Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 7 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 8 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 9 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Done

**Comment by jacoba109** [comment]
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=c5f982bf-3530-412b-a002-7ce560c3693c) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

File: dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java [hunk 12]
```diff
@@ -153,6 +153,6 @@ public void handleTask_multiArgText() throws Exception {
             throw e;
         }
 
-    }
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java:156]: fix whitespace

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 13]
```diff
@@ -0,0 +1,350 @@
...
+    //--------------------------------------------------------------------------
+    // Class-level setup / teardown
+    //--------------------------------------------------------------------------
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        server.startServer(true);
+
+        assertServerReady();
+        Log.info(thisClass, "setUpClass", "Server is ready?");
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: no need for log

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+
+        Log.info(thisClass, "setUpClass",
+                 "libertyInstallRoot=" + libertyInstallRoot + ", httpsPort=" + httpsPort
+                 + ", keystorePath=" + serverKeystorePath);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        if (server != null && server.isStarted()) {
+            // CWWKO0801E is expected
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: what is this warning

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 15]
```diff
@@ -0,0 +1,350 @@
...
+
+        // TLSv1.3 server — profiler must report successful handshakes
+        switchServerConfig(SERVER_XML_TLSv13);
+        ProgramOutput tlsv13Output = runTlsProfiler();
+
+        Log.info(thisClass, testName.getMethodName(), "TLSv1.3 stdout:\n" + tlsv13Output.getStdout());
+
+        assertEquals("TLSv1.3 server should succeed", SUCCESS_RC, tlsv13Output.getReturnCode());
+        assertTrue("stdout should contain the successful-handshakes header",
+                tlsv13Output.getStdout().contains("Successful handshakes"));
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: make more meaninful log finds

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+                    new String[] { "tlsProfiler", "--host=" + HOST, "--port=" + certServerPort },
+                    libertyInstallRoot, new Properties(), SHORT_TIMEOUT_SECS);
+
+            Log.info(thisClass, testName.getMethodName(),
+                     "RC: " + output.getReturnCode() + "\nstdout:\n" + output.getStdout() + "\nstderr:\n" + output.getStderr());
+
+            assertNotNull("Server should log CWWKO0801E with 'certificate_unknown' when the profiler connects without a trusted certificate",
+                          untrustedCertServer.waitForStringInLogUsingMark("CWWKO0801E.*certificate_unknown", 3000));
+        } finally {
+            untrustedCertServer.stopServer("CWWKO0801E");
```
Comment (rangaran) [inline dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:328]: why is there a warning

**Review by rangaran — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 16 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java:328]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 15 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Successful handshake doesn't indicate correct output, removed but kept other log finds as tlsProfiler hardcodes English output, expansion for other languages could be looked at in the future. Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 14 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Done

File: dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java (hunk 13 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility_fat/fat/src/com/ibm/ws/security/utility/test/SecurityUtilityTLSProfilerTest.java]: Done

File: dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java (hunk 12 above)
Comment (jacoba109) [reply dev/com.ibm.ws.security.utility/test/com/ibm/ws/security/utility/tasks/TLSProfilerTaskTest.java:156]: Done

**Comment by jacoba109** [comment]
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=ab10286d-6e15-40c2-b842-9907a04d5c4a) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

**Review by rangaran — APPROVED** [review]
